sklearn/compose/_target.py

```diff
@@ -127,6 +128,14 @@ class TransformedTargetRegressor(RegressorMixin, BaseEstimator):
     array([2.])
     """
 
+    _parameter_constraints = {
+        "regressor": [HasMethods("fit"), None],
```

Comment:
```suggestion
        "regressor": [HasMethods(["fit", "predict"]), None],
```

Reply:
Since we will call `predict` in `predict`, we can enforce the estimator to implement it as well.